src/bsp/tls/mbedtls/xi_bsp_sha256.c

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2003-2017, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client library,
+ * it is licensed under the BSD 3-Clause license.
+ */
+
+#include <xi_bsp_crypt.h>
+#include <xi_bsp_mem.h>
+#include <mbedtls/sha256.h>
+
+int xi_bsp_crypt_sha256_init( void** sha )
+{
+    *sha = xi_bsp_mem_alloc( sizeof( mbedtls_sha256_context ) );
+
+    mbedtls_sha256_init( ( mbedtls_sha256_context* )*sha );
+
+    return 0;
+}
+
+int xi_bsp_crypt_sha256_update( void* sha, const uint8_t* data, uint32_t len )
+{
+    mbedtls_sha256_starts( ( mbedtls_sha256_context* )sha, 0 );
```

Comment:
I don't think we should have function descriptions at call places. The documentation is at the definition.